webpack/scenes/ContentViews/Details/ComponentContentViews/ContentViewComponents.js

```diff
@@ -0,0 +1,114 @@
+import React, { useState, useEffect } from 'react';
+import { useSelector } from 'react-redux';
+import {
+  Bullseye,
+} from '@patternfly/react-core';
+import { Link } from 'react-router-dom';
+import { TableVariant, fitContent, TableText } from '@patternfly/react-table';
+import { STATUS } from 'foremanReact/constants';
+import { translate as __ } from 'foremanReact/common/I18n';
+import { urlBuilder } from 'foremanReact/common/urlHelpers';
+import PropTypes from 'prop-types';
+
+import TableWrapper from '../../../../components/Table/TableWrapper';
+import onSelect from '../../../../components/Table/helpers';
+import {
+  selectCVComponents,
+  selectCVComponentsStatus,
+  selectCVComponentsError,
+} from '../ContentViewDetailSelectors';
+import { getContentViewComponents } from '../ContentViewDetailActions';
+import AddedStatusLabel from '../../../../components/AddedStatusLabel';
+import ComponentVersion from './ComponentVersion';
+import ComponentEnvironments from './ComponentEnvironments';
+import ContentViewIcon from '../../components/ContentViewIcon';
+
+const ContentViewComponents = ({ cvId }) => {
+  const response = useSelector(state => selectCVComponents(state, cvId));
+  const status = useSelector(state => selectCVComponentsStatus(state, cvId));
+  const error = useSelector(state => selectCVComponentsError(state, cvId));
+
+  const [rows, setRows] = useState([]);
+  const [metadata, setMetadata] = useState({});
+  const [searchQuery, updateSearchQuery] = useState('');
+  const columnHeaders = [
+    { title: __('Type'), transforms: [fitContent] },
+    __('Name'),
+    __('Version'),
+    __('Environments'),
+    __('Repositories'),
+    { title: __('Status') },
+    __('Description'),
+  ];
+  const loading = status === STATUS.PENDING;
+
+  const buildRows = (results) => {
+    const newRows = [];
+    results.forEach((componentCV) => {
+      const { content_view: cv, content_view_version: cvVersion } = componentCV;
+      const { environments, repositories } = cvVersion;
```

Comment:
I got a blank page with this error in the console:
```
ContentViewComponents.js:49 Uncaught TypeError: Cannot read property 'environments' of null
    at ContentViewComponents.js:49
    at Array.forEach (<anonymous>)
    at buildRows (ContentViewComponents.js:47)
    at ContentViewComponents.js:77
    at commitHookEffectListMount (react-dom.development.js:19731)
    at commitPassiveHookEffects (react-dom.development.js:19769)
    at HTMLUnknownElement.callCallback (react-dom.development.js:188)
    at Object.invokeGuardedCallbackDev (react-dom.development.js:237)
    at invokeGuardedCallback (react-dom.development.js:292)
    at flushPassiveEffectsImpl (react-dom.development.js:22853)
```

I think it was because my component views hadn't been published yet.  Once I published a new version for each of my two component views, the error was gone.

I'd also like to see a test for this :)

Reply:
Updated.. Working on tests.

Reply:
Pushed a test to cover this..